common/network-shuffle/src/main/java/org/apache/spark/network/shuffle/RemoteBlockPushResolver.java

```diff
@@ -1382,6 +1395,7 @@ public void onData(String streamId, ByteBuffer buf) throws IOException {
           freeDeferredBufs();
```

Comment:
When it comes here because of `isTooLateBlockPush` then the whole block is going to be ignored. That means the deferredBufs which are getting freed will be ignored as well so we need to add those bytes to the ignoreBlockBytes  as well, correct?

Reply:
I think you are right. Have accommodated the suggestions below hence in this case it would be
`updateIgnoredBytes(deferredBytes + buf.remaining())`